src/main/java/org/wildfly/security/auth/login/SecurityDomain.java

```diff
@@ -47,21 +47,20 @@
  * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
  */
 public final class SecurityDomain {
-    private final Map<String, SecurityRealm> realmMap;
+    private final Map<String, RealmInfo> realmMap;
     private final String defaultRealmName;
-    private final NameRewriter[] preRealmRewriters;
+    private final NameRewriter preRealmRewriter;
     private final RealmMapper realmMapper;
-    private final NameRewriter[] postRealmRewriters;
+    private final NameRewriter postRealmRewriter;
     private final boolean anonymousAllowed;
     private final ThreadLocal<SecurityIdentity> currentSecurityIdentity = new ThreadLocal<>();
 
-    SecurityDomain(final Map<String, SecurityRealm> realmMap, final String defaultRealmName, final NameRewriter[] preRealmRewriters, final RealmMapper realmMapper, final NameRewriter[] postRealmRewriters) {
-        assert realmMap.containsKey(defaultRealmName);
-        this.realmMap = realmMap;
-        this.defaultRealmName = defaultRealmName;
-        this.preRealmRewriters = preRealmRewriters;
-        this.realmMapper = realmMapper;
-        this.postRealmRewriters = postRealmRewriters;
+    private SecurityDomain(Builder builder) {
```

Comment:
This should be package-private because it is called from the nested class.  When a constructor is private but it is accessed from a nested class, javac actually generates an invisible package-private constructor anyway.

Generally I only use private on methods and fields.  Otherwise everything else is public, protected, or package-private.